tests/Microsoft.Identity.Test.Unit/PublicApiTests/AuthenticationResultTests.cs

```diff
@@ -52,6 +55,111 @@ public void GetAuthorizationHeader()
             Assert.AreEqual("SomeTokenType at", ar.CreateAuthorizationHeader());
         }
 
+        [TestMethod]
+        public void GetIdTokenClaimsTest()
+        {
+            var ar = new AuthenticationResult(
+                "at",
+                false,
+                "uid",
+                DateTime.UtcNow,
+                DateTime.UtcNow,
+                "tid",
+                new Account("aid", "user", "env"),
+                MockHelpers.CreateIdTokenWithExtraClaim(TestConstants.UniqueId, TestConstants.DisplayableId, TestConstants.TenantId), 
+                new[] { "scope" }, Guid.NewGuid(),
+                "SomeTokenType",
+                new AuthenticationResultMetadata(TokenSource.Cache));
+            
+            ClaimsPrincipal claimsPrincipal = ar.GetIdTokenClaims();
+            Assert.IsNotNull(claimsPrincipal);
+            Assert.AreEqual(13, claimsPrincipal.Claims.Count());
+            var claims = claimsPrincipal.Claims;
+            IDictionary<string, string> claimsDictionary = new Dictionary<string, string>();
```

Comment:
i'm not sure what this is doing that's different from the code? seems like the same code repeated again...but maybe i'm reading this wrong.

Reply:
Here I am reversing the code to extract the claims to assert the claim I have added.